--- a/mlxlink/Makefile.am	1970-01-01 00:00:01.000000000 +0000
+++ b/mlxlink/Makefile.am	2024-08-02 03:30:50.762762140 +0000
@@ -68,9 +68,7 @@
                       $(LAYOUTS_DIR)/libtools_layouts.a \
                       $(USER_DIR)/xz_utils/libxz_utils.a \
                       $(USER_DIR)/ext_libs/minixz/libminixz.a \
-                      -lboost_regex -lboost_filesystem -lboost_system \
-                      -llzma $(LIBSTD_CPP) ${LDL} -lexpat \
                       $(JSON_LIBS)
 
-mstlink_LDADD = $(mstlink_DEPENDENCIES)
+mstlink_LDADD = $(mstlink_DEPENDENCIES) -lboost_filesystem -llzma $(LIBSTD_CPP) ${LDL} -lexpat
 
