From ac676d4aa419e0b7348ae5a2279bee76a114c0d7 Mon Sep 17 00:00:00 2001
From: natanbc <natanbc@users.noreply.github.com>
Date: Tue, 3 Jun 2025 04:02:34 +0000
Subject: [PATCH 2/2] speed up build

---
 Cargo.toml | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 807cf8d..6caa957 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,10 +13,11 @@ license = "Apache-2.0"
 panic = "abort"
 
 [profile.release]
-codegen-units = 1
-lto = true
+codegen-units = 16
+lto = false
 strip = true
 panic = "abort"
+opt-level = 2
 
 [profile.profiling]
 inherits = "release"
-- 
2.49.0

