diff --git a/Makefile b/Makefile
index c5988ca..1b6929a 100644
--- a/Makefile
+++ b/Makefile
@@ -5,8 +5,9 @@ endif
 # Or specific version
 #TARGET = 2.6.33.5

-KERNEL_MODULES = /lib/modules/$(TARGET)

+ifndef KERNEL_BUILD
+KERNEL_MODULES ?= /lib/modules/$(TARGET)
 ifneq ("","$(wildcard /usr/src/linux-headers-$(TARGET)/*)")
 # Ubuntu
 KERNEL_BUILD = /usr/src/linux-headers-$(TARGET)
@@ -18,16 +19,20 @@ else
 KERNEL_BUILD = $(KERNEL_MODULES)/build
 endif
 endif
+endif

 # SYSTEM_MAP = $(KERNEL_BUILD)/System.map
+ifndef SYSTEM_MAP
 ifneq ("","$(wildcard /boot/System.map-$(TARGET))")
 SYSTEM_MAP = /boot/System.map-$(TARGET)
 else
 # Arch
 SYSTEM_MAP = /proc/kallsyms
 endif
+endif

 DRIVER := it87
+ifndef DRIVER_VERSION
 ifneq ("","$(wildcard .git/*)")
 DRIVER_VERSION := $(shell git describe --long).$(shell date -u -d "$$(git show -s --format=%ci HEAD)" +%Y%m%d)
 else
@@ -37,6 +42,7 @@ else
 DRIVER_VERSION := unknown
 endif
 endif
+endif

 # DKMS
 DKMS_ROOT_PATH=/usr/src/$(DRIVER)-$(DRIVER_VERSION)
@@ -45,7 +51,9 @@ MODPROBE_OUTPUT=$(shell lsmod | grep it87)
 # Directory below /lib/modules/$(TARGET)/kernel into which to install
 # the module:
 MOD_SUBDIR = drivers/hwmon
+ifndef MODDESTDIR
 MODDESTDIR=$(KERNEL_MODULES)/kernel/$(MOD_SUBDIR)
+endif

 obj-m = $(patsubst %,%.o,$(DRIVER))
 obj-ko  := $(patsubst %,%.ko,$(DRIVER))
@@ -86,7 +94,7 @@ endif
 ifeq ($(COMPRESS_XZ), y)
	@xz -f $(MODDESTDIR)/$(DRIVER).ko
 endif
-	depmod -a -F $(SYSTEM_MAP) $(TARGET)
+	depmod -b $(MOD_ROOT) -a -F $(SYSTEM_MAP) $(TARGET)

 dkms:
	@mkdir -p $(DKMS_ROOT_PATH)
